src/main/java/kelbot/Parser.java

```diff
@@ -0,0 +1,93 @@
+package kelbot;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    private Command command;
+    private Integer taskNumber = Integer.MAX_VALUE;
+    private String keyword = "";
+    private String taskName = "";
+    private LocalDate date = null;
+    private boolean isValid = true;
+    /**
+     * Initializes Parser
+     *
+     * @param input The input given by the user
+     * @throws DateTimeParseException if the date is invalid
+     */
+    public Parser(String input) throws DateTimeParseException {
+        String[] commands = input.split(" ");
+        if (input.equals("bye")) {
+            command = Command.BYE;
+        } else if (input.equals("list")) {
+            command = Command.LIST;
+        } else if (commands[0].equals("done")) {
+            command = Command.DONE;
+            taskNumber = Integer.parseInt(commands[1]);
+        } else if (commands[0].equals("delete")) {
+            command = Command.DELETE;
+            taskNumber = Integer.parseInt(commands[1]);
+        } else if (commands[0].equals("find")) {
+            command = Command.FIND;
+            for (int i = 1; i < commands.length; i++) {
+                keyword += " " + commands[i];
+            }
+        } else if (commands[0].equals("todo")) {
+            command = Command.TODO;
+            for (int i = 1; i < commands.length; i++) {
+                taskName += " " + commands[i];
+            }
+        } else if (commands[0].equals("deadline")) {
+            command = Command.DEADLINE;
+            String deadlineDate = "";
+            for (int i = 1; i < commands.length; i++) {
+                if (commands[i].equals("/by")) {
+                    for (int j = i + 1; j < commands.length; j++) {
+                        deadlineDate += commands[j];
+                    }
+                    break;
+                }
+                taskName += " " + commands[i];
+            }
+            date = LocalDate.parse(deadlineDate);
+        } else if (commands[0].equals("event")) {
+            command = Command.EVENT;
+            String eventDate = "";
+            for (int i = 1; i < commands.length; i++) {
+                if (commands[i].equals("/at")) {
+                    for (int j = i + 1; j < commands.length; j++) {
+                        eventDate += commands[j];
+                    }
+                    break;
+                }
+                taskName += " " + commands[i];
+            }
+            date = LocalDate.parse(eventDate);
+        } else {
+            isValid = false;
+        }
+    }
+    public Command getCommand() {
```

Comment:
All these public methods not documented sia... are u planning to document them later maybe?

Reply:
I believe prof has mentioned that trivial methods do not require documentation!